### DISCUSSION TITLE
How do i make a snapshot main?

### DISCUSSION THREAD

**Discussion by diddiman** [Antynea/grub-btrfs#339] [Q&A]
So in grub i have my three recent kernels that i boot from normally. I also have the snapshots in submenu in grub
In btrfs i have /var as a subvolume so i can boot my snapshots in RO mode.

Imagine i boot into one snapshot, what is the easiest way to just apply that snapshot to my default boot entry?

So if i for instance boot normally, accidentally remove /etc/fstab or just break it. Restart and i can no longer boot. I go to my snapshot in RO mode. How do i copy everything in the snapshot to main? Or how do i just make this snapshot the default for my laptop? any best practices for this?

My guess is that if i for instance
```
❯ sudo btrfs subvolume list /
ID 256 gen 20641 top level 5 path home
ID 257 gen 20565 top level 5 path root
ID 258 gen 20147 top level 257 path var-old/lib/machines
ID 259 gen 14540 top level 257 path btrbk_snapshots/ROOT.20240713T1311
ID 264 gen 18645 top level 257 path btrbk_snapshots/ROOT.20240717T1859
ID 265 gen 18604 top level 257 path btrbk_snapshots/ROOT.20240717T1859_1
ID 266 gen 19752 top level 257 path btrbk_snapshots/ROOT.20240717T1915
ID 267 gen 19446 top level 257 path btrbk_snapshots/ROOT.20240717T1927
ID 268 gen 19444 top level 257 path btrbk_snapshots/ROOT.20240717T1959
ID 271 gen 20641 top level 257 path var
ID 272 gen 20318 top level 257 path btrbk_snapshots/ROOT.20240717T2207
```
boot into `ROOT.20240717T2207`, that i would simply just do a copy? So if i want to fully repair /etc/
I'd do `cp -rf /btrbk_snapshots/ROOT.20240717T2207/etc /`Is that correct? or am i missing something?

Thanks in advance!

**Comment by cjonesuk** [comment]
Did you work out the solution to this? i've just started using btrfs and grub-btrfs and i'm not sure how that works. @diddiman
